ompi/mca/osc/ucx/osc_ucx_comm.c

```diff
@@ -427,19 +453,42 @@ int ompi_osc_ucx_get(void *origin_addr, int origin_count,
                      int target, ptrdiff_t target_disp, int target_count,
                      struct ompi_datatype_t *target_dt, struct ompi_win_t *win) {
     ompi_osc_ucx_module_t *module = (ompi_osc_ucx_module_t*) win->w_osc_module;
-    ucp_ep_h ep = OSC_UCX_GET_EP(module->comm, target);
+    ucp_ep_h ep;
     uint64_t remote_addr = (module->win_info_array[target]).addr + target_disp * OSC_UCX_GET_DISP(module, target);
     ucp_rkey_h rkey;
     ptrdiff_t origin_lb, origin_extent, target_lb, target_extent;
     bool is_origin_contig = false, is_target_contig = false;
     ucs_status_t status;
+    pthread_t tid = pthread_self();
     int ret = OMPI_SUCCESS;
 
     ret = check_sync_state(module, target, false);
     if (ret != OMPI_SUCCESS) {
         return ret;
     }
 
+    if (pthread_equal(tid, mca_osc_ucx_component.main_tid)) {
```

Comment:
Maybe worth making this piece an inline function so it doesn't duplicate in put & get